backend/src/nodes/impl/normals/nvidia/LightspeedOctahedralConverter.py

```diff
@@ -0,0 +1,100 @@
+"""
+* SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+* SPDX-License-Identifier: MIT
+*
+* Permission is hereby granted, free of charge, to any person obtaining a
+* copy of this software and associated documentation files (the "Software"),
+* to deal in the Software without restriction, including without limitation
+* the rights to use, copy, modify, merge, publish, distribute, sublicense,
+* and/or sell copies of the Software, and to permit persons to whom the
+* Software is furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+* THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+* FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+* DEALINGS IN THE SOFTWARE.
+"""
+
+# This code has been modified from its original form
+
+import numpy as np
+from sanic.log import logger
+
+
+# Converts either OpenGL or DirectX style normal maps to RTX Remix compatible Hemispherical Octahedral maps.
+#
+# Note that normals pointing in to the surface are not physically possible, and are not supported by RTX Remix.
+#   Any images with inward pointing normals will generate a warning and will be flipped to point outwards.
+#
+# There is a good explanation of DirectX vs OpenGL normal maps at
+#   https://www.texturecan.com/post/3/DirectX-vs-OpenGL-Normal-Map/
+#
+# To then load these into RTX Remix, you can convert it to a DDS file using
+#   https://developer.nvidia.com/nvidia-texture-tools-exporter
+#   Use BC5 compression, and the flag --no-mip-gamma-correct
+class LightspeedOctahedralConverter:
+    # Convert DirectX style normal maps (green is down)
+    @staticmethod
+    def convert_dx_to_octahedral(image):
+        image = LightspeedOctahedralConverter._check_for_spherical_normals(image)
+        normals = LightspeedOctahedralConverter._pixels_to_normals(image)
```

Comment:
These 2 lines don't work, kinda. The problem is that they assume that the blue channel stores the z component of the normal as -1..1 mapping to 0..1 (`b = (z+1)/2`) (or 0..255 in the original implementation). This is somewhat uncommon. Since most game engines only support hemispherical normal maps (where z is between 0 and 1), those game engines don't encode negative z value (`b = abs(z)`). As we talked about before, z can be computed on the fly, which is why many game engines use the b channel to store other material information in there. So we can't assume anything about the b channel.

To fix this, we just have to ignore the blue channel and compute z on the fly. This is exactly what my `gr_to_xyz` util method does.